ui/src/pages/Questions/Ask/index.tsx

```diff
@@ -452,18 +448,18 @@ const Ask = () => {
                 {formData.content.errorMsg}
               </Form.Control.Feedback>
             </Form.Group>
+
             <Form.Group controlId="tags" className="my-3">
               <Form.Label>{t('form.fields.tags.label')}</Form.Label>
-              <Form.Control
-                defaultValue={JSON.stringify(formData.tags.value)}
-                isInvalid={formData.tags.isInvalid}
-                hidden
-              />
```

Comment:
![image](https://github.com/apache/incubator-answer/assets/30591202/29e06faa-1c1c-48f6-9524-7c1b5450dadb)

Correct error message：

![image](https://github.com/apache/incubator-answer/assets/30591202/c9290f5d-3c36-4baa-b8bc-0c8b440f6eae)

the tag error message is not displayed,there are several solutions: 
1. TagSelector component adds `custom-form-control is-invalid` class name
![image](https://github.com/apache/incubator-answer/assets/30591202/dc6b6dd4-c716-4199-8bc1-0523068ec627)
2. extend the component, put the error information of the component inside, and pass in the error information and status from props!

I prefer the second option！

Reply:
I got the issue that if classes ```custom-form-control``` and ```is-invalid``` both are not present in the child component of ```<Form.Group>``` component, the ```<Form.Control.Feedback>``` tag will not work. That's why it was not working earlier even though i passed ```is-invalid``` class because the ```is-invalid``` class was assign to its ```child's child``` component and ```<Form.Control.Feedback>``` has not been validated. 

![Screenshot 2024-05-07 170826](https://github.com/apache/incubator-answer/assets/43349097/3b279ee7-5692-4712-a32d-98f8be77b702)

But I didn't completely get your second approach.  
Should I suppose to do it this way
```<Form.Group controlId="tags" className="my-3">
  <Form.Label>{t('form.fields.tags.label')}</Form.Label>
  <TagSelector
    value={formData.tags.value}
    onChange={handleTagsChange}
    showRequiredTag
    maxTagLength={5}
   formData.tags
    }
  />
  <Form.Control.Feedback type="invalid">
    {formData.tags.errorMsg}
  </Form.Control.Feedback>
</Form.Group>
```
And then handle the error conditions inside ```TagSelector``` component?
I'm new to react, please help me to understand the expectations. Thanks you :)

Reply:
The cause of the problem is exactly as you described, so I say both solutions are possible.

In the second option, I just simplify the use of this component and build `<Form.Control.Feedback /> ` into the `TagSelector` component.

eg:
![image](https://github.com/apache/incubator-answer/assets/30591202/863a3aa1-c4c8-4bad-a20e-333c500aecae)

usage:
![image](https://github.com/apache/incubator-answer/assets/30591202/1ad259c0-8be6-4e52-8c65-b938208542f5)

Reply:
Understood!
Thanks for explaining @shuashuai.

But I have one concern about why we are putting ```<Form.Control.Feedback>``` component inside our custom ```<TagSelector>```. Every component is following this structure ...

```
<Form.Group controlId="tags" className="my-3">
  <Form.Label>title</Form.Label>                                // responsible for title of component
  <CustomComponent/>                                            // actual component
  <Form.Control.Feedback type="invalid">
    error                                                       // display error if any
  </Form.Control.Feedback>
</Form.Group>
```

If we follow the second approach for ```<TagSelector>```, then this structure is not followed. Should I make this change to all available components? Or, Can you tell me pros of following 2nd approach. Just want to get into details out of curiosity!

Reply:
This is indeed the case for a standard form structure as you mentioned.

I recommend the second option mainly from the perspective of ease of use of components, and unified management. From the first option, you can see that to combine this standard structure, we need to add additional attributes to the custom `<Form. Control />`  to conform to the standard structure. This may cause some confusion for some users who don’t know why.

For example, you may have found code like this:

![image](https://github.com/apache/incubator-answer/assets/30591202/d89ed79a-972d-4086-ac1f-45708938a7cb)

To conform to the bootstrap structure, it is easy to write such confusing code. At the same time, this may be done for components that use custom `form. control`, if the custom `form.control`  itself encapsulates the complete form function, it will greatly reduce the occurrence of this situation. it will also be easier to maintain.

The following structure will make the structure simpler and will not affect the reading of the code.

![image](https://github.com/apache/incubator-answer/assets/30591202/644e390a-8f68-4041-8bd7-d40f80f05c29)

Of course, this may be more in line with the coding habits of our team, and I also want to change these unusual writing methods in later versions.

Reply:
Understood!

Made changes to files, please verify.